### DISCUSSION TITLE
Problem importing CSV in TSML

### DISCUSSION THREAD

**Discussion by John-aa90** [code4recovery/12-step-meeting-list#944] [Q&A] [answered]
Hi, I'm not sure this is the right place for this request. My problem is that when I import a CSV file (exported from Meeting Guide)  in TSML, I end up with "Appointment" for the Time of every meeting. I checked and double-checked the format of "Time" and "End Time" fields. I played around with the format: leading zero or not, AM/PM or 24 hour, to no avail. I read that a "non-legit" name in the "day" field can cause the same error, is there some other fields that could produce a similar error? I tried importation on 2 different staging sites with the same results.
Test site: www.devtest.aa90.org
PHP 7.4.30
WP 6.0.2 fr_CA
TSML plugin 3.14.6

**Comment by joshreisner** [comment]
Hi @John-aa90 - I've moved this to a TSML discussion. Could you attach a screenshot or the first couple rows from your CSV? Sounds like TSML is not able to import the times properly from your spreadsheet.

**Reply by John-aa90** [reply]
![Screenshot from 2022-10-05 16-14-27](https://user-images.githubusercontent.com/115081276/194154863-f8492848-7559-43c7-a6c9-7ca16f9d1d02.png)
Hi @joshreisner - Here's a screenshot of the CSV's head. Thanks for your reply.

**Comment by joshreisner** [comment, marked as answer]
is your interface language in French? What if you change one of the days to Français? does that meeting import properly?

if that doesn't work perhaps could you share your whole CSV here? we can try importing it on our demo site

the double returns look a little suspect to me, but i can't say for sure if it's a problem or not. guessing it's ok if the meetings are importing ok otherwise

**Reply by John-aa90** [reply]
Hi Josh. My admin interface is in English but the sites are in French. My Usual browser is Chrome in English.  I have installed Firefox in French to detect translation problems, etc... By the way, TSML_UI works great to detect the language of the browser. I've  replaced all "Sunday" by "Dimanche" in the CSV and all the Sunday's meetings are displayed properly. I don't have time to do full testing right now but I'd say that the problem is solved. I didn't see this one coming because I assumed that the plugin would do bidirectional translation. Thank you very much for the fast response and for the wonderful job you've done with your apps and plugins

**Reply by John-aa90** [reply]
I don't see a place to mark this as solved

**Reply by joshreisner** [reply]
great! glad to hear it. i think the import page must be following the site language setting. we should make a note somewhere!

glad to hear TSML UI is handling language well. if you have any feedback on any of those translations please LMK in the TSML UI repo (where you first filed the ticket). i just used google translate for everything and i'm sure it's not optimal

**Comment by district6wc** [comment]
Hi I am having this issue with an English upload to an English site, I formatted the start and end times as time formats and the day is formatted as text. The file is actually an export from the plugin on our old site. I did this previously without an issue.

**Reply by joshreisner** [reply]
if you want to share your CSV we can try to import it, but my guess is it sounds like something went wrong when editing it, or saving after editing? we haven't heard any cases of exporting from one website and not being able to import to another website directly

**Comment by district6wc** [comment]
Josh

I saw another thread where someone mentioned using the csv UTF-8. I resaved
my file and it went up perfectly. Thanks for getting back. I couldn't do
this without all the guidance here.


Thank you.

Yours in Service,
Rose Lanaro
Saturday Morning Live, Chapel Hill

District 6 Website Coordinator Panel 62
631-525-3328
